syntax = "proto3";

package example;

service ExampleService {
  rpc ExchangeData(stream DataRequest) returns(stream DataResponse) {}
}

message DataRequest {
  bytes data = 1;
}

message DataResponse {
  bytes data = 1;
}

message SerializableData {
  string name = 1;
  double temperature = 2;
  int32 age = 3;
  bool gender = 4;
  bytes secret = 5;
}